pkgs/games/prismlauncher/wrapper.nix

```diff
@@ -21,10 +25,16 @@
 , msaClientID ? null
 , gamemodeSupport ? stdenv.isLinux
 , textToSpeechSupport ? stdenv.isLinux
+, waylandSupport ? false
+, shellWrapper ? waylandSupport
```

Comment:
It's there to show how it's used. I think having `shellWrapper ? waylandSupport` is a nice visual.

Reply:
> It isn't really a redundancy, it serves a purpose. The assertion is there for good reason: while not necessary for the wrapper to function, it is for the reader figuring out how this works.

if `shellWrapper` had a use on it's own outside of documentation i'd agree, but i think an addition to the comment for `waylandSupport` would be better suited for this while being just as effective at getting the message across

Reply:
I will not remove the assertion.

Reply:
i will not be approving this pr then. i see no reason to have a user facing option that only adds runtime costs, especially when a comment - something that could provide much more information at zero cost - could be used in its place.

if you have a better reason as to why an assertion better helps in "figuring out how this works" than a comment, please let me know. in the meantime, "i will not remove the assertion" with no further elaboration or compromise will be a blocker

Reply:
I said I wouldn't remove the assertion, not that I wouldn't remove the argument.

The assertion makes it very clear that `makeWrapper` is required, and the comment explains why.

Do you think a reasonable compromise would be to put the `shellWrapper` in a let block (with comment above) and keep the assertion?

Reply:
while i would still argue the benefit of having a variable over a comment (especially when evaluation time is considered), i would be fine with that compromise since it isn't really user facing anymore and we could revisit this without breaking any overrides people have